### ISSUE TITLE
fail to reproduce the result

### ISSUE THREAD

**Issue by yuzekai1234** [Trae1ounG/DyPRAG#19]
when use llama3-8b-instruct to reproduce the result, i find the translator still does not converge，even after training with 800 pieces of data. 
Stage 2: Epoch 0, Step 811, MSE Loss: 0.0048, LM Loss: 1.9032, KL Loss: 8.9685, Target LM Loss: 0.8445
Stage 2: Epoch 0, Step 812, MSE Loss: 0.0049, LM Loss: 2.0466, KL Loss: 9.1203, Target LM Loss: 0.9617
Stage 2: Epoch 0, Step 813, MSE Loss: 0.0047, LM Loss: 1.9718, KL Loss: 8.5843, Target LM Loss: 0.8877
Stage 2: Epoch 0, Step 814, MSE Loss: 0.0034, LM Loss: 1.8875, KL Loss: 8.5107, Target LM Loss: 0.8600
Stage 2: Epoch 0, Step 815, MSE Loss: 0.0027, LM Loss: 1.9204, KL Loss: 9.7912, Target LM Loss: 0.7680
Stage 2: Epoch 0, Step 816, MSE Loss: 0.0511, LM Loss: 2.0669, KL Loss: 10.0617, Target LM Loss: 0.9709
Stage 2: Epoch 0, Step 817, MSE Loss: 0.0023, LM Loss: 2.0286, KL Loss: 8.8181, Target LM Loss: 0.9026
Stage 2: Epoch 0, Step 818, MSE Loss: 0.0038, LM Loss: 1.9168, KL Loss: 9.4788, Target LM Loss: 0.8474
Stage 2: Epoch 0, Step 819, MSE Loss: 0.0216, LM Loss: 1.9570, KL Loss: 9.5885, Target LM Loss: 0.9181

**Comment by Trae1ounG** [comment]
Do you include any modification? In my log, the translator converge quickly and only a little oscillation between samples.

**Comment by yuzekai1234** [comment]
I didn’t change the core code, but I used your training data prepared for Llama 3.2-1B-Instruct to train a translator for Llama 3.1-8B-Instruct. Is that the core problem?
python3 -u src/train_dyprag.py     --model_name=llama3.1-8b-instruct     --datasets="2wikimultihopqa"     --learning_rate=0.0003     --lora_rank=2     --lora_alpha=32     --max_new_tokens=128     --sample_rate=1     --dyprag_learning_rate=1e-5     --dyprag_train_epochs=1 \

**Comment by Trae1ounG** [comment]
Maybe you can include more datasets like
```
--datasets="2wikimultihopqa,hotpotqa,popqa,complexwebquestions" \
```
As the translator with diverse resources data will converge easily.

**Comment by yuzekai1234** [comment]
so if i only use wikimultihopqa,the model will not converge?Have you ever tested it?

**Comment by Trae1ounG** [comment]
You can try test on it. The loss metric does not fully reflect the final performance.
